internal/services/manageddevopspools/managed_devops_pool_resource_test.go

```diff
@@ -0,0 +1,266 @@
+	r := ManagedDevOpsPoolResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (ManagedDevOpsPoolResource) Exists(ctx context.Context, client *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := pools.ParsePoolID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := client.ManagedDevOpsPools.PoolsClient.Get(ctx, *id)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving %s: %+v", *id, err)
+	}
+
+	return utils.Bool(resp.Model != nil), nil
+}
+
+func (r ManagedDevOpsPoolResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_managed_devops_pool" "test" {
+  name                = "acctest-pool-%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+
+  maximum_concurrency            = 1
+  dev_center_project_resource_id = azurerm_dev_center_project.test.id
+
+  organization_profile {
+    kind = "AzureDevOps"
+    organizations {
+      parallelism = 1
+      url         = "https://dev.azure.com/managed-org-demo"
+    }
+  }
+
+  agent_profile {
+    kind = "Stateless"
+  }
+
+  fabric_profile {
+    kind = "Vmss"
+    images {
+      resource_id = "/Subscriptions/e21f7bce-1728-44ad-a62d-344064a0d69a/Providers/Microsoft.Compute/Locations/australiaeast/publishers/canonical/artifacttypes/vmimage/offers/0001-com-ubuntu-server-focal/skus/20_04-lts-gen2/versions/latest"
+      buffer      = "*"
+    }
+    sku {
+      name = "Standard_D2ads_v5"
+    }
+  }
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (r ManagedDevOpsPoolResource) requiresImport(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_managed_devops_pool" "import" {
+  name                = azurerm_managed_devops_pool.test.name
+  location            = azurerm_managed_devops_pool.test.location
+  resource_group_name = azurerm_managed_devops_pool.test.resource_group_name
+
+  maximum_concurrency            = 1
+  dev_center_project_resource_id = azurerm_dev_center_project.test.id
+
+  organization_profile {
+    kind = "AzureDevOps"
+    organizations {
+      parallelism = 1
+      url         = "https://dev.azure.com/managed-org-demo"
+    }
+  }
+
+  agent_profile {
+    kind = "Stateless"
+  }
+
+  fabric_profile {
+    kind = "Vmss"
+    images {
+      resource_id = "/Subscriptions/e21f7bce-1728-44ad-a62d-344064a0d69a/Providers/Microsoft.Compute/Locations/australiaeast/publishers/canonical/artifacttypes/vmimage/offers/0001-com-ubuntu-server-focal/skus/20_04-lts-gen2/versions/latest"
```

Comment:
The testcases will run in the Hashicorp's testing subscriptions, can this resource will be used?

Reply:
updated to use the data.client().subscriptoin and these are compute images from marketplace available on every subscription.